### DISCUSSION TITLE
Advice for making the chat window having a distinct and polished look

### DISCUSSION THREAD

**Discussion by csheaff** [chep/copilot-chat.el#60] [Q&A]
Thanks for this package. It works well. 

One obstacle I find myself encountering when using this package as opposed to, say, the VScode implementation, is the readability.  Another is managing both buffers when using the markdown/org front ends together as i open up/close other buffers. 

I imagine the functionality/extensibility is there to make something really polished looking, but I'm not sure how to go about it. I am currently using the shell-maker front end as it is only a single buffer, which is easier to manage. I've tried to see how i can spruce up the shell-maker buffer with not much luck.

Any advice would be appreciated.

**Comment by chep** [comment]
Hi,
I don't know the VSCode plugin but I know readability is the main problem. org-mode is the best frontend I think for this point but is not perfect.
I'm trying to find a way to bind buffers/windows together, but for the moment I don't know what to do at all.
There are a lot of packages to help managing windows that could help you: ace-window, emacs-purpose…

I'll try VSCode and find some ways to improve copilot-chat.el.

Any help is welcome.

**Comment by pace-gene** [comment]
I also have the same problem. I particularly dislike the separate prompt buffer/window. I'll have a look if I've any spare time if the input and output can somehow be combined into one in the same way they do it in Zed for example.

@chep for reference, here's a screenshot of the assistant pane in Zed (very similar to vscode):
![image](https://github.com/user-attachments/assets/90c0a017-0720-4089-bea7-bbdb47637737)

**Comment by chep** [comment]
I would also like to get something like this. Perhaps by drawing inspiration from the ERC code: the last line serves as a prompt, and the rest of the buffer is for the data.
This undoubtedly represents a lot of work to integrate this system with the main modes. I want to keep them, especially org-mode, which has many features for content manipulation and offers a lot of possibilities.

**Comment by chep** [comment]
Polymode looks very promising.

**Comment by chep** [comment]
I pushed a new branch : polymode.

I'm gonna test it to while finishing developping this feature.

For the moment, it breaks everything except basic chat with org frontend.

There is only one window with prompt at its bottom. The prompt separator is to be defined, for the moment it has been chosen by copilot itself.

Please try this branch and tell me what you think about it.

**Reply by pace-gene** [reply]
Wow, this is already much better than the previous two-window setup. I've noticed that the point tends to not jump to the next prompt, but that's a minor annoyance. I'll keep using this branch and see how I fare.

**Reply by chep** [reply]
I merged it into master.

**Comment by csheaff** [comment]
Great work. Thanks a lot!
